RecoMET/METAlgorithms/interface/GlobalHaloAlgo.h

```diff
@@ -71,13 +74,80 @@ class GlobalHaloAlgo {
+  void setRcaloMinRsegmLowThresholdforCSCCaloMatchingEB(float x ){dr_lowthresh_segvsrh_eb= x;}
+  void setRcaloMinRsegmHighThresholdforCSCCaloMatchingEB(float x ){dr_highthresh_segvsrh_eb= x;}
+  void setDtcalosegmThresholdforCSCCaloMatchingEB(float x ){dt_segvsrh_eb= x;}
+  void setDPhicalosegmThresholdforCSCCaloMatchingEB(float x ){dphi_thresh_segvsrh_eb= x;}
+  //EE  
+  void setEtThresholdforCSCCaloMatchingEE(float x ){et_thresh_rh_ee= x;}
+  void setRcaloMinRsegmLowThresholdforCSCCaloMatchingEE(float x ){dr_lowthresh_segvsrh_ee= x;}
+  void setRcaloMinRsegmHighThresholdforCSCCaloMatchingEE(float x ){dr_highthresh_segvsrh_ee= x;}
+  void setDtcalosegmThresholdforCSCCaloMatchingEE(float x ){dt_segvsrh_ee= x;}
+  void setDPhicalosegmThresholdforCSCCaloMatchingEE(float x ){dphi_thresh_segvsrh_ee= x;}
+  //HB  
+  void setEtThresholdforCSCCaloMatchingHB(float x ){et_thresh_rh_hb= x;}
+  void setRcaloMinRsegmLowThresholdforCSCCaloMatchingHB(float x ){dr_lowthresh_segvsrh_hb= x;}
+  void setRcaloMinRsegmHighThresholdforCSCCaloMatchingHB(float x ){dr_highthresh_segvsrh_hb= x;}
+  void setDtcalosegmThresholdforCSCCaloMatchingHB(float x ){dt_segvsrh_hb= x;}
+  void setDPhicalosegmThresholdforCSCCaloMatchingHB(float x ){dphi_thresh_segvsrh_hb= x;}
+  //HE  
+  void setEtThresholdforCSCCaloMatchingHE(float x ){et_thresh_rh_he= x;}
+  void setRcaloMinRsegmLowThresholdforCSCCaloMatchingHE(float x ){dr_lowthresh_segvsrh_he= x;}
+  void setRcaloMinRsegmHighThresholdforCSCCaloMatchingHE(float x ){dr_highthresh_segvsrh_he= x;}
+  void setDtcalosegmThresholdforCSCCaloMatchingHE(float x ){dt_segvsrh_he= x;}
+  void setDPhicalosegmThresholdforCSCCaloMatchingHE(float x ){dphi_thresh_segvsrh_he= x;}
+
+
   
  private:
   float Ecal_R_Min;
   float Ecal_R_Max;
   float Hcal_R_Min;
   float Hcal_R_Max;
   float TowerEtThreshold;
+
+  //Parameters for CSC-calo matching
+  float max_segment_theta;
+
+  float  et_thresh_rh_eb;
+  float dphi_thresh_segvsrh_eb;
+  float dr_lowthresh_segvsrh_eb;
+  float dr_highthresh_segvsrh_eb;
+  float dt_segvsrh_eb;
+
+  float  et_thresh_rh_ee;
+  float dphi_thresh_segvsrh_ee;
+  float dr_lowthresh_segvsrh_ee;
+  float dr_highthresh_segvsrh_ee;
+  float dt_segvsrh_ee;
+
+  float  et_thresh_rh_hb;
+  float dphi_thresh_segvsrh_hb;
+  float dr_lowthresh_segvsrh_hb;
+  float dr_highthresh_segvsrh_hb;
+  float dt_segvsrh_hb;
+
+  float  et_thresh_rh_he;
+  float dphi_thresh_segvsrh_he;
+  float dr_lowthresh_segvsrh_he;
+  float dr_highthresh_segvsrh_he;
+  float dt_segvsrh_he;
+
+
+
+  void AddtoBeamHaloEBEERechits(edm::RefVector<EcalRecHitCollection>& bhtaggedrechits,reco::GlobalHaloData & thehalodata, bool isbarrel);
+  void AddtoBeamHaloHBHERechits(edm::RefVector<HBHERecHitCollection>& bhtaggedrechits,reco::GlobalHaloData & thehalodata);
+  bool SegmentMatchingEB(reco::GlobalHaloData & thehalodata, std::vector<reco::HaloClusterCandidateECAL>haloclustercands, float iZ, float iR, float iT, float iPhi, bool ishlt);
```

Comment:
for better performance, pass ``` std::vector<reco::HaloClusterCandidateECAL> ``` by const reference here and in the remaining methods.
The code appears to be reasonably fast already; so, change not required, but nice to have.